conseil-api/src/test/scala/tech/cryptonomic/conseil/api/routes/platform/data/ethereum/EthereumDataRoutesTest.scala

```diff
@@ -0,0 +1,211 @@
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import tech.cryptonomic.conseil.api.metadata.{
+  AttributeValuesCacheConfiguration,
+  MetadataService,
+  TransparentUnitTransformation
+}
+import tech.cryptonomic.conseil.api.routes.platform.discovery.TestPlatformDiscoveryOperations
+import tech.cryptonomic.conseil.common.config.MetadataConfiguration
+import tech.cryptonomic.conseil.common.config.Platforms.{EthereumConfiguration, PlatformsConfiguration}
+import tech.cryptonomic.conseil.common.generic.chain.DataTypes.{Query, QueryResponse}
+import tech.cryptonomic.conseil.common.generic.chain.PlatformDiscoveryTypes.{Attribute, DataType, Entity, KeyType}
+import tech.cryptonomic.conseil.common.metadata.{EntityPath, NetworkPath, PlatformPath}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class EthereumDataRoutesTest
+    extends AnyWordSpec
+    with Matchers
+    with ScalatestRouteTest
+    with ScalaFutures
+    with MockFactory
+    with BeforeAndAfterEach
+    with EthereumDataRoutesTest.Fixtures {
+
+  private val testEntity = Entity("blocks", "Test Entity", 0)
+  private val testNetworkPath = NetworkPath("mainnet", PlatformPath("ethereum"))
+  private val testEntityPath = EntityPath("blocks", testNetworkPath)
+  private val platformDiscoveryOperations = new TestPlatformDiscoveryOperations
+  platformDiscoveryOperations.addEntity(testNetworkPath, testEntity)
+  blockAttributes.foreach(platformDiscoveryOperations.addAttribute(testEntityPath, _))
+
+  private val conseilOps: EthereumDataOperations = new EthereumDataOperations("ethereum") {
+    override def queryWithPredicates(prefix: String, tableName: String, query: Query)(
+        implicit ec: ExecutionContext
+    ): Future[List[QueryResponse]] =
+      Future.successful(responseAsMap)
+  }
+
+  private val metadataService =
+    new MetadataService(
+      PlatformsConfiguration(
+        List(
+          EthereumConfiguration(
+            "mainnet",
+            enabled = true
+          )
+        )
+      ),
+      TransparentUnitTransformation,
+      stub[AttributeValuesCacheConfiguration],
+      platformDiscoveryOperations
+    )
+  private val routes: EthereumDataRoutes =
+    EthereumDataRoutes(metadataService, MetadataConfiguration(Map.empty), conseilOps, 1000)
+
+  "Query protocol" should {
+    "return a correct response with OK status code with POST" in {
+
+      val postRequest = HttpRequest(
+        HttpMethods.POST,
+        uri = "/v2/data/ethereum/mainnet/blocks",
+        entity = HttpEntity(MediaTypes.`application/json`, jsonStringRequest)
+      )
+
+      postRequest ~> addHeader("apiKey", "hooman") ~> routes.postRoute ~> check {
+        val resp = entityAs[String]
+        resp.filterNot(_.isWhitespace) shouldBe jsonStringResponse.filterNot(_.isWhitespace)
```

Comment:
I think there is a scalatest plugin lib to compare json strings for match, that we already use.
It's more robust to how the json is formatted, and should check the content instead of the string.

Reply:
Right, you are taking probably about `matchJson` from custom library for scalatest. Find that, so it's added already.